src/services/invitation/index.ts

```diff
@@ -55,6 +57,38 @@ const plugin: FastifyPluginAsync = async (fastify, options) => {
     },
   );
 
+  fastify.register(fastifyMultipart, {
+    limits: {
+      fields: MAX_NON_FILE_FIELDS, // Max number of non-file fields (Default: Infinity).
+      fileSize: MAX_FILE_SIZE, // For multipart forms, the max file size (Default: Infinity).
+      files: MAX_FILES, // Max number of file fields (Default: Infinity).
+    },
+  });
+
+  fastify.route<{ Querystring: IdParam; Body: any }>({
+    method: HttpMethod.POST,
+    url: '/:id/invitations/upload_csv',
+    preHandler: fastify.verifyAuthentication,
+    handler: async (request, reply) => {
+      const { member, query, log } = request;
+      const errors: Error[] = [];
+      try {
+        const file = await request.file();
+        await db.transaction(async (manager) => {
+          await iS.handleCSVInvitations(member, buildRepositories(manager), query, file);
+        });
+      } catch (e) {
+        console.log(e);
+        errors.push(e);
+      }
+      return {
+        // TO-DO: report invitations in data
+        data: [],
```

Comment:
This return shape is what we call a `ResultOf<T>`. It is used when we perform a "many" operation, for example asking for multiple memberships at a time. We use this shape to be able to reply with a partial set of the data and maybe some errors. In your case, I think that the operation should not be considered partial, it should either succeed or fail. So you should simply return the data if it succeeded and an error if it failed.

Reply:
The issue becomes very apparent when you see that you will always at most push one error in your error array, which will just mean that the whole thing failed.